### PR TITLE
Add a new QML category "Geometry Options"

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/core/qgsvectorlayer.cpp [hunk 1]
```diff
@@ -2064,11 +2064,12 @@ bool QgsVectorLayer::readSymbology( const QDomNode &layerNode, QString &errorMes
...
   }
 
+  if ( categories.testFlag( GeometryOptions ) )
+    mGeometryOptions->readXml( layerNode.namedItem( QStringLiteral( "geometryOptions" ) ) );
+
   if ( categories.testFlag( LayerConfiguration ) )
   {
     QDomElement geometryOptionsElement = layerNode.namedItem( QStringLiteral( "geometryOptions" ) ).toElement();
-    mGeometryFixes->setGeometryPrecision( geometryOptionsElement.attribute( QStringLiteral( "geometryPrecision" ),  QStringLiteral( "0.0" ) ).toDouble() );
-    mGeometryFixes->setRemoveDuplicateNodes( geometryOptionsElement.attribute( QStringLiteral( "removeDuplicateNodes" ),  QStringLiteral( "0" ) ).toInt() == 1 );
```
Comment (wonder-sk) [inline src/core/qgsvectorlayer.cpp]: Leftover ```QDomElement geometryOptionsElement``` ?

File: src/core/qgsvectorlayer.cpp [hunk 2]
```diff
@@ -2286,13 +2287,11 @@ bool QgsVectorLayer::writeSymbology( QDomNode &node, QDomDocument &doc, QString
...
 
-  if ( categories.testFlag( LayerConfiguration ) )
-  {
-    QDomElement geometryOptionsElement = doc.createElement( QStringLiteral( "geometryOptions" ) );
-    node.appendChild( geometryOptionsElement );
-    geometryOptionsElement.setAttribute( QStringLiteral( "removeDuplicateNodes" ), mGeometryFixes->removeDuplicateNodes() ? 1 : 0 );
-    geometryOptionsElement.setAttribute( QStringLiteral( "geometryPrecision" ), mGeometryFixes->geometryPrecision() );
-  }
+  QDomElement fieldConfigurationElement = doc.createElement( QStringLiteral( "fieldConfiguration" ) );
+  node.appendChild( fieldConfigurationElement );
```
Comment (wonder-sk) [inline src/core/qgsvectorlayer.cpp]: Is the addition of empty "fieldConfiguration" element needed/wanted?

File: src/core/qgsgeometryoptions.h [hunk 3]
```diff
@@ -83,6 +85,38 @@ class CORE_EXPORT QgsGeometryFixes
...
+     * \since QGIS 3.4
+     */
+    QStringList geometryChecks() const;
+
+    /**
+     * A list of activated geometry checks.
+     *
+     * \since QGIS 3.4
+     */
+    void setGeometryChecks( const QStringList &geometryChecks );
```
Comment (wonder-sk) [inline src/core/qgsgeometryoptions.h]: So how are the geometry checks going to work? Can we have an enum with the available checks? It is difficult for users of the API to figure out what checks can be used and what are the valid configuration options...

**Comment by m-kuhn** [comment]
@wonder-sk give me half an hour, I've got a followup pending which I push as soon as the cache is warm from the current build.

**Review by 3nids — APPROVED** [review]

**Comment by m-kuhn** [comment]
@wonder-sk the geometry check configuration options will come back with a followup pull request with a bit more context to discuss.

**Pull request merged**